docs/features/enumerations.rst

```diff
@@ -0,0 +1,217 @@
+Enumerations
+============
+
+Description
+-----------
+
+DSG can handle the generation of enumerations in your `.proto` files in three ways:
+
+- Using a Serializer or Model field with TextChoices (limited features)
+- Using an Annotated Serializer or Model field (all features)
+- Using FieldDict in the `grpc_action` decorator (all features)
+
+Unless you are using enums in a FieldDict directly, you should be using `TextChoices` or `IntegerChoices` as defined in the Django documentation `here <https://docs.djangoproject.com/en/5.1/ref/models/fields/#enumeration-types>`_.
+
+Using a FieldDict
+-----------------
+
+In a FieldDict, you can specify an Enum for the type. This will generate the corresponding enum in the `.proto` file.
+
+.. code-block:: python
+
+    @grpc_action(
+        request=[{"name": "enum_example", "type": MyEnum}],
+        response=[{"name": "value", "type": "string"}],
+    )
+
+Here is the generated enum in `.proto` file:
+
+.. code-block:: protobuf
+
+    message Request {
+        MyEnum.Enum enum_example = 1;
+    }
+
+    message MyEnum {
+        enum Enum {
+            ENUM_UNSPECIFIED = 0;
+            FIRST = 1;
+            SECOND = 2;
+        }
+    }
+
+Using a Serializer or Model field with TextChoices
+--------------------------------------------------
+
+This method is limited to the use of `TextChoices` in Django models or serializers and only include a subset of the features, but it is the simplest way to generate an enum.
+
+For this to work your TextChoices need to be defined with key-like members in the first value of the tuple (letters all in uppercase, numbers allowed after the first letter, and underscore character).
+
+This works :
+
+.. code-block:: python
+
+    class MyModel(models.Model):
+        class MyEnum(models.TextChoices):
+            FIRST = "FIRST", "First"
+            #        ^^^^^  This will be the first key of the enum
+            SECOND = "SECOND", "Second"
+
+        my_field = models.CharField(choices=MyEnum.choices)
+
+Here is the generated enum in .proto file:
+
+.. code-block:: protobuf
+
+    message Request {
+        MyEnum.Enum my_field = 1;
+    }
+
+    message MyEnum {
+        enum Enum {
+            ENUM_UNSPECIFIED = 0;
+            FIRST = 1;
+            SECOND = 2;
+        }
+    }
+
+This doesn't work and will be generated as a string field:
+
+.. code-block:: python
+
+    class MyModel(models.Model):
```

Comment:
why is this example not working ?

Reply:
Because the field is not annotated, and in that case the first values of the tuple should be key like members as explained above

Reply:
I stil dont' understand the issue with that sorry. What is the blocking point ?

Reply:
You can see in that model that the field is not annotated, which means that we will lose access to the keys of the enums when accessing the field, because django normalise the choices.
Because of this we have to check if we can dynamically create an enum using the values of our TextChoices.

This is the enum : 
        class MyEnum(models.TextChoices):
            FIRST = "First", "First"
            SECOND = "SECOND VALUE", "Second"

MyEnum.choices return this, and that's all we have access to, when creating the enum on the fly : 
("First", "First"),
("SECOND VALUE", "Second")

in the first value, "First" we are not all capitalized, and in the second "SECOND VALUE" we miss a underscore instead of a space
-> we can't build an enum

Reply:
You cannot acces the key of the enum ?

Following the django doc you have: A number of custom properties are added to the enumeration classes – .choices, .labels, .values, and .names – to make it easier to access lists of those separate parts of the enumeration.

So you could have FIRST and SECOND as key for the enum and "First" , "SECOND VALUE" for the possible value

Reply:
(in 4.0 you could pass the whole TextChoices, but it was not meant to be used like that, in 4.0 doc they pass MyTextChoices.choices)